mlflow/models/model.py

```diff
@@ -109,6 +127,21 @@ def saved_input_example_info(self, value: Dict[str, Any]):
         # pylint: disable=attribute-defined-outside-init
         self._saved_input_example_info = value
 
+    def get_model_info(self) -> ModelInfo:
```

Comment:
Do we need this method? Can we just add `model_uri` as an attribute of `Model`?

Reply:
We can add `model_uri` as a property of `Model` like:

@property
def model_uri():
  return artifact_utils.get_artifact_uri(self.run_id, self.artifact_path)

Reply:
@harupy After offline discussions, we decide to use `ModelInfo` instead of `Model` for the following reasons:
1. `Model` contains many other methods that do not make sense in the return value, e.g. `add_flavor(...)` since `Model` is mutable while the return value should be immutable.
2. `ModelInfo` k-v contents can be printed since it's implemented as a `namedtuple`, while `Model` k-v contents cannot be printed out-of-the-box.

```
m = mlflow.xxx_yyy.log_model(...)
m.model_uri
```
is the current usage pattern, where m is a `ModelInfo` instance.